Adding nickname searching in database search

vincedbowen opened the pull request:
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #4352 

<!-- Concisely describe what the pull request does. -->
Allows users to search their databases using the database nicknames as well as the database name. 

**Technical details**
The solution simply concatenates the two strings together in the form: <db_name db_nickname>. If this is an undesirable solution, please let me know, and I can change the backend datastructure from a string to an array of strings or something similar.

**Screenshots**
The name of the database can still be searched on 
![Screenshot 2025-03-18 at 5 50 06 PM](https://github.com/user-attachments/assets/3b0cfa89-8ced-482c-aee2-416e3cdcaca8)
Now, the nickname of the database can also be searched on
![Screenshot 2025-03-18 at 5 50 40 PM](https://github.com/user-attachments/assets/b7675d03-b8ca-4933-8f74-6542318ce9a5)

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

seancolsen reviewed (approved):
Thanks for this PR @vincedbowen!

I added e04ac1b815bfa52cb76067233fcc52f617c58b87 and 8a733462170a21bcb474a135f38fd3a68b0d044e with some small cleanup to address a few things:

- Your PR would not have passed CI due to Prettier not being run. You can read more about this [here](https://github.com/mathesar-foundation/mathesar/blob/develop/mathesar_ui/README.md#formatting). Be sure to do this for any PRs you open in the future.

- The string concatenation you've used here is clever. I can see that this mostly works. But it's a tad hacky for my taste.

    We'd like to search in multiple strings, so I'd rather pass multiple strings in an array instead of together as one string.

    The string concatenation introduced a subtle bug too... If I have a database named "books" with no nickname, searching for the string "ul" would incorrectly display my books database as a match. Can you guess why? Because the nickname field, being `null`, gets converted to the string "null" and thus becomes part of the searchable text.

    Using arrays allow us to leverage the type system to avoid this pitfall. TS will complain at you if you put `d.nickname` directly in the array — hence the extra `allNames()` method on `Database` to bundle up the complexity of retrieving one or both names for the Database.

vincedbowen commented:
> I added [e04ac1b](https://github.com/mathesar-foundation/mathesar/commit/e04ac1b815bfa52cb76067233fcc52f617c58b87) and [8a73346](https://github.com/mathesar-foundation/mathesar/commit/8a733462170a21bcb474a135f38fd3a68b0d044e) with some small cleanup to address a few things:
> 
>     * Your PR would not have passed CI due to Prettier not being run. You can read more about this [here](https://github.com/mathesar-foundation/mathesar/blob/develop/mathesar_ui/README.md#formatting). Be sure to do this for any PRs you open in the future.
> 
>     * The string concatenation you've used here is clever. I can see that this mostly works. But it's a tad hacky for my taste.
>       We'd like to search in multiple strings, so I'd rather pass multiple strings in an array instead of together as one string. I've fixed this.
>       The string concatenation introduced a subtle bug too... If I have a database named "books" with no nickname, searching for the string "ul" would incorrectly display my books database as a match. Can you guess why? Because the nickname field, being `null`, gets converted to the string "null" and thus becomes part of the searchable text.
>       Using arrays allow us to leverage the type system to avoid this pitfall. TS will complain at you if you put `d.nickname` directly in the array — hence the extra `allNames()` method on `Database` to bundle up the complexity of retrieving one or both names for the Database.

1. Will do from now on :-). Apologies for missing that. 
2. I figured it wasn't a production quality solution, but didn't even think of that bug, so thanks for catching that!
